coalib/coala_main.py

```diff
@@ -86,11 +86,14 @@ def run_coala(console_printer=None,
                                   VERSION))
 
         settings_hash = get_settings_hash(sections, targets)
-        flush_cache = bool(sections['default'].get('flush_cache', False) or
-                           settings_changed(log_printer, settings_hash))
+        flush_cache = bool(
+            (sections['cli'] or sections['default']).get(
+                'flush_cache', False) or
+            settings_changed(log_printer, settings_hash))
 
         cache = None
-        if not sections['default'].get('disable_caching', False):
+        if not (sections['cli'] or sections['default']).get(
+                'disable_caching', False):
```

Comment:
uhm does this or statement really work? I mean if the cli section doesn't exist or is empty it will either raise an exception or it'll just take the default (False), this doesn't look too good to me. Maybe even make a function `get_cli_setting(key, type, default)`?